### DISCUSSION TITLE
Correct animation direction for workspaces sorted by name

### DISCUSSION THREAD

**Discussion by aserowy** [hyprwm/Hyprland#3828] [Q&A] [answered]
Heho,

i'm using hyprland with waybar. On waybar the workspaces are sorted by name. My workflow follows this order. Thus, hitting ctrl+n e.g. focuses the next workspace by name. The animation on the other hand takes only the order of creation to get the direction from where the workspace comes in.

My question: Is there a way to configure the behavior and i'm just too stupid to find it? Or is there any other way to force this behavior by setting the id (e.g. hashing the name) and the name while creating a workspace?

Kind regards and thanks for this hypr cool wm!
Alexander

**Comment by vaxerski** [comment]
Not really, hyprland sorts the workspaces by ID, and those IDs are created newest to the left, oldest to the right for named.

**Reply by aserowy** [reply]
I saw, that i can specify the id while creating a workspace. Is there a way to set id and name in the same command?

**Reply by aserowy** [reply]
Heho, after the rename in waybar gets handled correctly, i worked on a str to int resolution to set sortable ids. The method works but:

hyprctl dispatch workspace 9223372036854775806

is not working and gives Err: stoi, although the documentation on https://wiki.hyprland.org/Configuring/Dispatchers/#workspaces gives this id as inlusive max. The dispatch accepts only numbers with a length of 9 currently. Am i missing something?

**Reply by vaxerski** [reply, marked as answer]
my bad, docs are wrong. Max is 2147483647

**Reply by aserowy** [reply]
@vaxerski you were fast.. was correcting the wiki and swooop, nothing to correct anymore :D

**Reply by aserowy** [reply]
if anyone wants the same behavior:

the id gets resolved with the following script. I create the workspace with the given id and rename it afterwards.

```nu
def get_workspace_id_by_name [workspaces: list<any>, name: string] {
    mut min = 1;
    mut max = 2147483647;

    let sorted = ($workspaces
        | select id name
        | where id > 0
        | append {id: 0, name: $name}
        | sort-by name --ignore-case --natural)
    
    mut switch = true;
    for ws in $sorted {
        if $ws.id == 0 {
            $switch = false
        } else if $switch {
            if $min < $ws.id {
                $min = $ws.id
            }
        } else {
            if $max > $ws.id {
                $max = $ws.id
            }
        }
    }

    return (($min + $max) / 2 | into int)
}
```
